static/app/utils/replays/extractDomNodes.tsx

```diff
@@ -17,187 +13,126 @@ export type Extraction = {
 };
 
 type Args = {
-  finishedAt: Date | undefined;
   frames: (BreadcrumbFrame | SpanFrame)[] | undefined;
   rrwebEvents: RecordingFrame[] | undefined;
 };
 
-function _extractDomNodes({
-  frames,
+export default function extractDomNodes({
+  frames = [],
   rrwebEvents,
-  finishedAt,
 }: Args): Promise<Extraction[]> {
-  // Get a list of the BreadcrumbFrames that relate directly to the DOM, for each
-  // frame we will extract the referenced HTML.
-  if (!frames || !rrwebEvents || rrwebEvents.length < 2 || !finishedAt) {
-    return Promise.reject();
-  }
-
-  return new Promise((resolve, reject) => {
-    const domRoot = document.createElement('div');
-    domRoot.className = 'sentry-block';
-    const {style} = domRoot;
-    style.position = 'fixed';
-    style.inset = '0';
-    style.width = '0';
-    style.height = '0';
-    style.overflow = 'hidden';
-
-    document.body.appendChild(domRoot);
-
-    // Grab the last event, but skip the synthetic `replay-end` event that the
-    // ReplayerReader added. RRWeb will skip that event when it comes time to render
-    const lastEvent = rrwebEvents[rrwebEvents.length - 2];
-
-    const isLastRRWebEvent = (event: RecordingFrame) => lastEvent === event;
-
-    const replayerRef = new Replayer(rrwebEvents, {
-      root: domRoot,
-      loadTimeout: 1,
-      showWarning: false,
-      blockClass: 'sentry-block',
-      speed: 99999,
-      skipInactive: true,
-      triggerFocus: false,
-      plugins: [
-        new BreadcrumbReferencesPlugin({
-          frames,
-          isFinished: isLastRRWebEvent,
-          onFinish: rows => {
-            resolve(rows);
-            setTimeout(() => {
-              if (document.body.contains(domRoot)) {
-                document.body.removeChild(domRoot);
-              }
-            }, 0);
-          },
-        }),
-      ],
-      mouseTail: false,
-    });
-
-    try {
-      // Run the replay to the end, we will capture data as it streams into the plugin
-      replayerRef.pause(finishedAt.getTime());
-    } catch (error) {
-      Sentry.captureException(error);
-      reject(error);
-    }
-  });
-}
+  return new Promise(resolve => {
+    const extractions = new Map<BreadcrumbFrame | SpanFrame, Extraction>();
+    const player = createPlayer(rrwebEvents);
+    let lastEventTimestamp = 0;
+
+    const callback = event => {
+      if (event.type === 2 || event.type === 3) {
```

Comment:
I think we should be able to use `import {EventType} from 'sentry/utils/replays/types';` instead of integers `2` and `3`